src/algorithms/SortingAlgorithms.java

```diff
@@ -3,45 +3,136 @@
 public class SortingAlgorithms {
 
     public static void main(String[] args) {
-        int [] array = new int[]{2,1,8,3,5,5,-26};
-        sortByBubbleSorting(array);
-        print(array);
+        int[] array = new int[] {2, 12, 3, 21, 8, 0, 18, -26};
+//           sortByBubbleSorting(array);
+//          print(array, "sortByBubbleSorting");
+//         sortBySelectionSorting(array);
+//        print(array, "sortBySelectionSorting");
+//       sortByInsertionSorting(array);
+//      print(array, " sortByInsertionSorting");
+//     sortByMergeSorting(array);
+//    print(array, "sortByMergeSorting");
+   sortByQuickSorting(array,0,array.length-1);
+  print(array, "sortByQuickSorting");
     }
 
     public static void sortByBubbleSorting(int[] array) {
         for (int i = 0; i < array.length; i++) {
-            for (int j = 0; j < array.length-i-1; j++) {
-                if (array[j] > array[j+1]) {
-                    swap(array, j+1, j);
+            for (int j = 0; j < array.length - i - 1; j++) {
+                if (array[j] > array[j + 1]) {
+                    swap(array, j + 1, j);
                 }
             }
         }
     }
 
-    public static void sortBySelectionSorting(int[] array){
+    public static void sortBySelectionSorting(int[] array) {
+        for (int k = 0; k < array.length; k++) {
+            for (int l = k + 1; l < array.length; l++) {
+                if (array[l] < array[k]) {
+                    swap(array, k, l);
+                }
+            }
+        }
+    }
 
+    public static void sortByInsertionSorting(int[] array) {
+        for (int k =0; k < array.length; ) {
+            for (int l = k + 1; l < array.length; l++) {
+                if (array[l] < array[k]) {
+                    swap(array,k,l);
+                       int m=k;
+                  while(m>0 && array[m-1]>array[m]){
+                    swap(array,m-1,m);
+                      m--;
+                  }
+                }
+                 k++;
```

Comment:
A for loop stop condition should test the loop counter against an invariant value (i.e. one that is true at both the beginning and ending of every loop iteration). Ideally, this means that the stop condition is set to a local variable just before the loop begins.
Stop conditions that are not invariant are slightly less efficient, as well as being difficult to understand and maintain, and likely lead to the introduction of errors in the future.